public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -341,15 +150,14 @@ export class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogs
           <div className="gf-form gf-form--grow flex-shrink-1">
             <QueryField
               additionalPlugins={this.plugins}
-              query={(query as CloudWatchLogsQuery).expression ?? ''}
+              query={expression ?? ''}
               onChange={this.onChangeQuery}
-              onClick={this.onQueryFieldClick}
               onRunQuery={this.props.onRunQuery}
               onTypeahead={this.onTypeahead}
               cleanText={cleanText}
               placeholder="Enter a CloudWatch Logs Insights query (run with Shift+Enter)"
               portalOrigin="cloudwatch"
-              disabled={loadingLogGroups || selectedLogGroups.length === 0}
+              disabled={logGroupNames?.length === 0}
```

Comment:
Think you want to do `!logGroupNames?.length` because if logGroupNames is undefined this expression will be falsy.